shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Mempool.hs

```diff
@@ -189,9 +263,14 @@ overNewEpochState ::
   NewEpochState era ->
   f (NewEpochState era)
 overNewEpochState f st = do
-  res <- f $ mkMempoolState st
-  pure $
-    st
-      { LedgerState.nesEs =
-          (LedgerState.nesEs st) {LedgerState.esLState = res}
-      }
+  f (mkMempoolState st)
```

Comment:
Minor: Unless I'm thinking too quickly here, you only need `Functor f`, not `Applicative f`. (IE it's just a "lens", not a "traversal".)